src/main/java/ru/practicum/shareit/item/service/ItemServiceImpl.java

```diff
@@ -19,67 +28,173 @@
+                .map(CommentMapper::toCommentDto)
+                .collect(Collectors.toList());
+
+        ItemWithBookingsDto itemDto = ItemMapper.toItemWithBookingDto(item,
+                null,
+                null,
+                comments);
+
+
+        if (item.getOwner().getId() != userId) {
+            return itemDto;
+        }
+
+        String status = "APPROVED";
+        LocalDateTime now = LocalDateTime.now();
+        Booking lastBooking = bookingRepository.getLastBookingByItem(itemId, now, status);
+        Booking nextBooking = bookingRepository.getNextBookingByItem(itemId, now, status);
+
+        if (lastBooking != null) itemDto.setLastBooking(BookingMapper.toBookingWithIdAndBookerId(lastBooking.getId(),
+                lastBooking.getBooker().getId()));
+
+        if (nextBooking != null) itemDto.setNextBooking(BookingMapper.toBookingWithIdAndBookerId(nextBooking.getId(),
+                nextBooking.getBooker().getId()));
+
+        return itemDto;
     }
 
-    public Item get(Long itemId) {
-        if (!itemStorage.getAllItems().contains(itemStorage.get(itemId))) {
-            log.debug("Такой вещи не существует");
-            throw new NotFoundException("Такой вещи не существует");
+    @Override
+    public List<ItemWithBookingsDto> getAllItemsByUser(Long userId) {
+
+        checkUsrId(userId);
+
+        String status = "APPROVED";
+        LocalDateTime now = LocalDateTime.now();
+        List<Item> itemList = itemRepository.findAllByOwnerId(userId);
+
+        if (itemList.isEmpty()) {
+            log.debug("У вас нет вещей");
+            return new ArrayList<>();
         }
 
-        return itemStorage.get(itemId);
+        List<ItemWithBookingsDto> newList = new ArrayList<>();
+
+        for (Item i : itemList) {
+            Booking lastBooking = bookingRepository.getLastBookingByItem(i.getId(), now, status);
+            Booking nextBooking = bookingRepository.getNextBookingByItem(i.getId(), now, status);
+            List<CommentDto> comments = commentRepository.findAllByItemId(i.getId()).stream()
+                    .map(CommentMapper::toCommentDto)
+                    .collect(Collectors.toList());
+
+            ItemWithBookingsDto item = ItemMapper.toItemWithBookingDto(i,
+                    null,
+                    null,
+                    comments);
+
+            if (lastBooking != null) item.setLastBooking(BookingMapper.toBookingWithIdAndBookerId(lastBooking.getId(),
+                    lastBooking.getBooker().getId()));
+
+            if (nextBooking != null) item.setNextBooking(BookingMapper.toBookingWithIdAndBookerId(nextBooking.getId(),
+                    nextBooking.getBooker().getId()));
+
+            newList.add(item);
+        }
+        return newList;
     }
 
-    public List<ItemDto> getAll(Long userId) {
+    @Override
+    public List<ItemDto> getRequired(String text) {
+        if (text.isBlank()) return Collections.emptyList();
```

Comment:
Даже такие маленькие if лучше оформлять как :
```suggestion
        if (text.isBlank()) 
        {
            return Collections.emptyList();
         }
```
Так очевиднее